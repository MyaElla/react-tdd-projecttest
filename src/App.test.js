import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';
// 
// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });


const AppC = shallow(<App />)

it('renders without crashing', () => {
const div = document.createElement('div');
ReactDOM.render(<App />, div);
ReactDOM.unmountComponentAtNode(div);
expect(AppC).toMatchSnapshot()
})
