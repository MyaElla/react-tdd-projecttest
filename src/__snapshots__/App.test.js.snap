// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          My Product Store
        </h1>,
        <p>
          You have selected 
          0
           product(s).
        </p>,
        <ProductList
          onProductSelect={[Function]}
          products={
            Array [
              Object {
                "brand": "Nike",
                "id": 1,
                "name": "AirMax 90",
              },
              Object {
                "brand": "Adidas",
                "id": 2,
                "name": "Yeezy",
              },
              Object {
                "brand": "Reebok",
                "id": 3,
                "name": "Classic",
              },
            ]
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "My Product Store",
        },
        "ref": null,
        "rendered": "My Product Store",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "You have selected ",
            0,
            " product(s).",
          ],
        },
        "ref": null,
        "rendered": Array [
          "You have selected ",
          0,
          " product(s).",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onProductSelect": [Function],
          "products": Array [
            Object {
              "brand": "Nike",
              "id": 1,
              "name": "AirMax 90",
            },
            Object {
              "brand": "Adidas",
              "id": 2,
              "name": "Yeezy",
            },
            Object {
              "brand": "Reebok",
              "id": 3,
              "name": "Classic",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            My Product Store
          </h1>,
          <p>
            You have selected 
            0
             product(s).
          </p>,
          <ProductList
            onProductSelect={[Function]}
            products={
              Array [
                Object {
                  "brand": "Nike",
                  "id": 1,
                  "name": "AirMax 90",
                },
                Object {
                  "brand": "Adidas",
                  "id": 2,
                  "name": "Yeezy",
                },
                Object {
                  "brand": "Reebok",
                  "id": 3,
                  "name": "Classic",
                },
              ]
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My Product Store",
          },
          "ref": null,
          "rendered": "My Product Store",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "You have selected ",
              0,
              " product(s).",
            ],
          },
          "ref": null,
          "rendered": Array [
            "You have selected ",
            0,
            " product(s).",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onProductSelect": [Function],
            "products": Array [
              Object {
                "brand": "Nike",
                "id": 1,
                "name": "AirMax 90",
              },
              Object {
                "brand": "Adidas",
                "id": 2,
                "name": "Yeezy",
              },
              Object {
                "brand": "Reebok",
                "id": 3,
                "name": "Classic",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
